blockchain/types/transaction_test.go

```diff
@@ -633,13 +635,21 @@ func TestIntrinsicGas(t *testing.T) {
 		assert.Equal(t, tc.expectGas2, gas)
 		assert.Equal(t, nil, err)
 
-		gas, err = IntrinsicGas(data, nil, true, params.Rules{IsIstanbul: false})
+		gas, err = IntrinsicGas(data, nil, false, params.Rules{IsPrague: true})
```

Comment:
Just to be safe..
```suggestion
		gas, err = IntrinsicGas(data, nil, false, params.Rules{IsIstanbul: true, IsPrague: true})
```

Reply:
To be safer, how bout `params.Rules{IsIstanbul: true, IsShanghai: true, IsPrague: true}`

Reply:
@blukat29
The numbers were exactly fitted for `Prague` HF test.
```
nonZeroGas = params.TxDataNonZeroGasEIP2028
zeroGas = params.TxDataZeroGas
```

Can you explain in more detail how adding `IsIstanbul` contributes to the context safe?


@ian0371 The `expectedGas*` numbers did not consider the `Shanghai` value. Do you mean to add this case additionally?

Reply:
It's because if a block is Prague, it implies it's Istanbul and Shanghai as well.
In other words, a block can't be Prague if it's not Istanbul and not Shanghai.

Hence, these are only valid rules:
```
params.Rules{IsIstanbul: true}
params.Rules{IsIstanbul: true, IsShanghai: true}
params.Rules{IsIstanbul: true, IsShanghai: true, IsPrague: true}
```

Reply:
That makes sense. Then, do you suggest refining the `expectGas*` values? FYI. Without changing this, the test fails if other HFs are added, as you mentioned.

Reply:
Then please adjust the testcases to reflect the reality

Reply:
Updated https://github.com/kaiachain/kaia/pull/107/commits/c94555e18d200cd1eafbf5beec078c2991aff76d